subprojects/docs/src/docs/userguide/migration/upgrading_version_6.adoc

```diff
@@ -38,6 +38,20 @@ Some plugins will break with this new version of Gradle, for example because the
 
 === Potential breaking changes
 
+==== Removal of `compile` and `runtime` configurations
+
+Since its inception, Gradle provided the `compile` and `runtime` configurations to declare dependencies.
+Alternatives were introduced in Gradle 3.4:
+
+- the `implementation` configuration should be used to declare dependencies which are _implementation details_ of a library: they will not leak to consumers when themselves are built
+- the `api` configuration, available only if you apply the `java-library` plugin, should be used to declare dependencies which are part of the API of a component, that is to say exposed to consumers
+
+In Gradle 7, both the `compile` and `runtime` configurations were removed.
+Therefore, you have to migrate to the `implementation` and `api` configurations above.
+If you are still using the `java` plugin for a Java library, you will need to apply the `java-library` plugin instead.
+
+You can find more details about the benefits of the new configurations and which one to use in place of `compile` and `runtime` by reading the <<java_library_plugin.adoc#java_library_plugin,Java Library plugin>> documentation.
```

Comment:
I am really not convinced we should recommend to use `api`. The whole point of the change is to make people choose between `api` and `implementation`.

Reply:
I would argue for adding something like I proposed with your argument above:

> ...if you are frustrated that the configurations are removed.

In that case you might not care about all of this right now. Maybe you want to upgrade to 7 and do not want to address api/impl separation right now. In that cases I think this is a valid way forward. You can then upgrade to 7, and decide to do better api/impl separation later. Ending this section with "please read java_library_plugin and figure out what to do on your own" sounds like it would frustrate such users even more.

It does not mean everyone should/will do it. We offer both alternatives here.

Reply:
Wouldn't this use case be better covered by a "backport plugin"?

Reply:
Maybe if we end up offering something like that. You may leave this like it is now, but we should get back and see if we can improve the advice here if we offer something more for backward compatibility.